packages/@uppy/companion-client/types/index.d.ts

```diff
@@ -1,48 +1,71 @@
+/**
+ * Async storage interface, similar to `localStorage`. This can be used to
+ * implement custom storages for authentication tokens.
+ */
+export interface TokenStorage {
+  setItem: (key: string, value: string) => Promise<void>
+  getItem: (key: string) => Promise<string>
+  removeItem: (key: string) => Promise<void>
+}
+
 export interface RequestClientOptions {
-  companionUrl: string;
-  companionHeaders?: object;
+  companionUrl: string
+  companionHeaders?: object
   /**
    * Deprecated, use `companionHeaders` instead.
    */
-  serverHeaders?: object;
+  serverHeaders?: object
 }
 
 export class RequestClient {
-  constructor (uppy: any, opts: RequestClientOptions);
-  get (path: string): Promise<any>;
-  post (path: string, data: object): Promise<any>;
-  delete (path: string, data: object): Promise<any>;
+  constructor (uppy: any, opts: RequestClientOptions)
+  get (path: string): Promise<any>
+  post (path: string, data: object): Promise<any>
+  delete (path: string, data: object): Promise<any>
 }
 
-export interface ProviderOptions extends RequestClientOptions {
-  provider: string;
-  authProvider?: string;
-  name?: string;
+/**
+ * Options for Providers that can be passed in by Uppy users through
+ * Plugin constructors.
+ */
+export interface PublicProviderOptions extends RequestClientOptions {
+  companionAllowedHosts?: string | RegExp | Array<string | RegExp>
+  storage?: TokenStorage
```

Comment:
apologies for the mix up here 🙈, but based on [this PR](https://github.com/transloadit/uppy/pull/2046/files) which I submitted recently, it became apparent that the `storage` option is only supplied to the provider plugins themselves (i.e Dropbox, Instagram, etc.). A fallback is set for the provider plugins [like so](https://github.com/transloadit/uppy/blob/master/packages/%40uppy/companion-client/src/Provider.js#L99). It is then used by `companion-client/Provider` class [like so](https://github.com/transloadit/uppy/blob/master/packages/%40uppy/companion-client/src/Provider.js#L42). In essence, what I think we need to do is remove this line ☝️ and then add the field [here](https://github.com/transloadit/uppy/pull/1918/files#diff-8c1b43bfbf1e79e2aab309c4f195090dR10) (same for Instagram, OneDrive, etc.). Apologies for the mx up once again

Reply:
ahh makes sense. no worries